network/p2p/libp2pNode.go

```diff
@@ -486,89 +181,6 @@ func (n *Node) CreateStream(ctx context.Context, peerID peer.ID) (libp2pnet.Stre
 	return stream, nil
 }
 
-// tryCreateNewStream makes at most `maxAttempts` to create a stream with the peer.
-// This was put in as a fix for #2416. PubSub and 1-1 communication compete with each other when trying to connect to
-// remote nodes and once in a while NewStream returns an error 'both yamux endpoints are clients'.
-//
-// Note that in case an existing TCP connection underneath to `peerID` exists, that connection is utilized for creating a new stream.
-// The multiaddr.Multiaddr return value represents the addresses of `peerID` we dial while trying to create a stream to it.
-func (n *Node) tryCreateNewStream(ctx context.Context, peerID peer.ID, maxAttempts int) (libp2pnet.Stream, []multiaddr.Multiaddr, error) {
```

Comment:
moved this logic to the unicast manager, since it is shared for creating all types of streams, e.g., default, compressed, etc.